web/src/components/users/AdminUserEditForm.vue

```diff
@@ -0,0 +1,307 @@
+          :readonly="isNil(groupMembershipAttributes.divisionId)"
+          :append-inner-icon="isNil(groupMembershipAttributes.divisionId) ? 'mdi-lock' : ''"
+          @update:model-value="updateBranch"
+        />
+      </v-col>
+      <v-col
+        cols="12"
+        md="6"
+      >
+        <UserGroupAutocomplete
+          :model-value="groupMembershipAttributes.unitId"
+          :type="UserGroupTypes.UNIT"
+          :parent-id="groupMembershipAttributes.branchId"
+          label="Unit"
+          variant="outlined"
+          clearable
+          :readonly="isNil(groupMembershipAttributes.branchId)"
+          :append-inner-icon="isNil(groupMembershipAttributes.branchId) ? 'mdi-lock' : ''"
+          @update:model-value="updateUnit"
+        />
+      </v-col>
+    </v-row>
+
+    <v-row>
+      <v-col
+        cols="12"
+        md="6"
+      >
+        <h3 class="mb-1">Roles</h3>
+
+        <v-chip
+          v-for="(roleType, index) in user.roleTypes"
+          :key="index"
+          class="ma-2"
+          color="info"
+          size="large"
+        >
+          {{ formatRole(roleType) }}
+        </v-chip>
+      </v-col>
+    </v-row>
+    <v-row>
+      <v-spacer />
+      <v-col class="d-flex justify-end">
+        <v-btn
+          :loading="isLoading"
+          color="error"
+          variant="outlined"
+          v-bind="cancelButtonOptions"
+          text="Cancel"
+        />
+        <!-- TODO: add ability to request update of yukon government directory information during save -->
+        <v-btn
+          class="ml-3"
+          :loading="isLoading"
+          type="submit"
+          color="success"
+          text="Save"
+        />
+      </v-col>
+    </v-row>
+  </v-form>
+</template>
+
+<script setup lang="ts">
+import { isNil } from "lodash"
+import { ref, toRefs, watch } from "vue"
+import { useI18n } from "vue-i18n"
+import { VBtn } from "vuetify/lib/components/index.mjs"
+
+import { required } from "@/utils/validators"
+
+import { GroupMembership } from "@/api/users-api"
+import { UserGroupTypes } from "@/api/user-groups-api"
+import useSnack from "@/use/use-snack"
+import useUser from "@/use/use-user"
+
+import UserGroupAutocomplete from "@/components/user-groups/UserGroupAutocomplete.vue"
+
+type CancelButtonOptions = VBtn["$props"]
+
+const props = withDefaults(
+  defineProps<{
+    userId: number
+    cancelButtonOptions?: CancelButtonOptions
+  }>(),
+  {
+    cancelButtonOptions: ({ userId }) => ({
+      to: {
+        name: "UserPage",
+        params: { userId },
+      },
+    }),
+  }
+)
+
+const emit = defineEmits(["saved"])
+
+const { userId } = toRefs(props)
+const { user, save, isLoading } = useUser(userId)
+
+const snack = useSnack()
+
+const isValid = ref(false)
+const groupMembershipAttributes = ref<Partial<GroupMembership>>({})
+
+async function saveWrapper() {
+  if (!isValid.value) {
+    snack.notify("Please fill out all required fields", { color: "error" })
+    return
+  }
```

Comment:
Interesting, I've been using `form.value?.validate()`, lately, and I'm thinking I just forgot that form v-model was a thing? I'll have to double check, since this version is a lot cleaner!